simulator/risc_v/riscv_instr.cpp

```diff
@@ -122,99 +159,168 @@ struct RISCVTableEntry
+    {'C', instr_c_addiw,    execute_c_addiw<I>,    OUT_ARITHM, ImmediateType::C_ADDIW,    Imm::ARITH, Src1::ZERO,       Src2::ZERO,       Dst::RD,         0, RV64_RV128},
+    // Constant-Generation
+    {'C', instr_c_li,       execute_c_li<I>,       OUT_ARITHM, ImmediateType::C_LI,       Imm::ARITH, Src1::ZERO,       Src2::ZERO,       Dst::RD,         0},
+        /* If Dst == 2 for instr_c_lui, then it is instr_c_addi16sp */
+    {'C', instr_c_lui,      execute_c_lui<I>,      OUT_ARITHM, ImmediateType::C_LUI,      Imm::LOGIC, Src1::ZERO,       Src2::ZERO,       Dst::RD,         0},
+    // Immediate logic and comparison
+    {'C', instr_c_andi,     execute_c_andi<I>,     OUT_ARITHM, ImmediateType::C_ANDI,     Imm::ARITH, Src1::ZERO,       Src2::ZERO,       Dst::RS1_3_BITS, 0},
+    // Register-register arithmetics
+    {'C', instr_c_sub,      execute_c_sub<I>,      OUT_ARITHM, ' ',                       Imm::NO,    Src1::ZERO,       Src2::RS2_3_BITS, Dst::RS1_3_BITS, 0},
+    {'C', instr_c_addw,     execute_c_addw<I>,     OUT_ARITHM, ' ',                       Imm::NO,    Src1::ZERO,       Src2::RS2_3_BITS, Dst::RS1_3_BITS, 0, RV64_RV128},
+    {'C', instr_c_subw,     execute_c_subw<I>,     OUT_ARITHM, ' ',                       Imm::NO,    Src1::ZERO,       Src2::RS2_3_BITS, Dst::RS1_3_BITS, 0, RV64_RV128},
+        /* If Src2 == 0 for instr_c_mv, then it is instr_c_jr */
+    {'C', instr_c_mv,       execute_c_mv<I>,       OUT_ARITHM, ' ',                       Imm::NO,    Src1::ZERO,       Src2::RS2_COMPR,  Dst::RD,         0},
+        /* if Src2 == 0 && Dst == 0 for instr_c_add, then it is instr_c_ebreak */
+        /* if Src2 == 0 for instr_c_add, then it is instr_c_jalr */
+    {'C', instr_c_add,      execute_c_add<I>,      OUT_ARITHM, ' ',                       Imm::NO,    Src1::ZERO,       Src2::RS2_COMPR,  Dst::RD,         0},
+    // Register-register logic and comparison
+    {'C', instr_c_xor,      execute_c_xor<I>,      OUT_ARITHM, ' ',                       Imm::NO,    Src1::ZERO,       Src2::RS2_3_BITS, Dst::RS1_3_BITS, 0},
+    {'C', instr_c_or,       execute_c_or<I>,       OUT_ARITHM, ' ',                       Imm::NO,    Src1::ZERO,       Src2::RS2_3_BITS, Dst::RS1_3_BITS, 0},
+    {'C', instr_c_and,      execute_c_and<I>,      OUT_ARITHM, ' ',                       Imm::NO,    Src1::ZERO,       Src2::RS2_3_BITS, Dst::RS1_3_BITS, 0},
 };
 
 template<typename I>
 static const RISCVTableEntry<I> invalid_instr =
 {'I', {"unknown", 0, 0}, &ALU::unknown_instruction<I>, OUT_ARITHM, ' ', Imm::NO, Src1::ZERO, Src2::ZERO, Dst::ZERO, 0};
 
 template<typename I>
-const auto& find_entry( uint32 bytes)
+const auto& find_entry( uint32 bytes, uint16 base)
 {
     for (const auto& e : cmd_desc<I>)
-        if ( e.entry.check_mask( bytes))
-            return e;
-        
+        if ( e.entry.check_mask( bytes)) {
+            switch(e.base)
+            {
+                case RV_ANY:     return e;
```

Comment:
We do not need to check whether the instruction is correct for that mode or not, since we trust the source of instructions completely. Once throwing an `illegal exception` is necessary, we'll think about more accurate solution.

Reply:
So how is it better to resolve the problem of instructions with same encodings? E.g.: c_addiw and c_jal.

Reply:
Oh, I did not know that. Then leave it as is and address other comments related (sizeof etc.)